runtime/autoload/health/provider.vim

```diff
@@ -387,29 +387,37 @@ function! s:check_python(version) abort
 
   let pip = 'pip' . (a:version == 2 ? '' : '3')
 
-  if !empty(python_exe)
+  if empty(python_exe)
+    " No Python executable can import 'neovim'. Check if any Python executable
+    " can import 'pynvim'. If so, that Python failed to import 'neovim' as
+    " well, which is most probably due to a failed pip upgrade:
+    " https://github.com/neovim/neovim/wiki/Following-HEAD#20181118
+    let [pynvim_exe, errors] = provider#pythonx#DetectByModule('pynvim', a:version)
+    if !empty(pynvim_exe)
+      call health#report_error(
+            \ 'Detected pip upgrade failure: Python executable can import "pynvim" but '
+            \ . 'not "neovim": '. pynvim_exe,
+            \ "Use that Python version to reinstall \"pynvim\" and optionally \"neovim\".\n"
+            \ . pip ." uninstall pynvim neovim\n"
+            \ . pip ." install pynvim\n"
+            \ . pip ." install neovim  # only if needed by third-party software")
+    endif
+  else
```

Comment:
@mhinz 
This code does not handle using a host var properly: `python_exe` might be empty (e.g. because "check_bin" rejects it, https://github.com/neovim/neovim/pull/11047), and then this code will use it as-is (without extra checks being done for `python_exe`), resulting in a false positive.

This could be fixed by using `provider#pythonx#CheckForModule(exe, a:module, a:major_version)` directly when "python_exe" comes via the host var.  (but it needs to be re-retrieved from there then; likely a helper var might be usedful, since the from-host-var-check is done several times)

Reply:
Yeah, please go for it. You seem to be more into that code right now. :)

Reply:
Maybe this whole help-with-pip-upgrade check should be removed?

At least it should maybe also use "python -m pip" style instead of "pip", especially when it comes to issues it tries to report, where pip in PATH might be different from python.

Reply:
Well, we certainly won't need that check forever, but we added it because a lot of people had that issue when `neovim` was renamed to `pynvim`: https://github.com/neovim/neovim/wiki/Following-HEAD#20181118

It's up to @justinmk to decide when to remove it again. :]

As for `pip` vs. `python -m pip`: I don't even know the effective difference without looking it up, so.. yeah, I'd rather care about the Ruby side.

Reply:
> At least it should maybe also use "python -m pip" style instead of "pip",

Reply:
@justinmk 
Cool.
But given that the upgrade migration period is over by now already, do you think we should rather drop this check, instead of fixing it for two different issues?
(it also simplifies code!)

Reply:
Is it over? Some people still use nvim 0.1.7, this stuff takes years.

Reply:
I've meant the transition to the new Python package name.

Reply:
Yes, my point is that it's likely there are still many people who haven't upgraded their python packages.